apps/ensadmin/src/hooks/ensnode-connections.tsx

```diff
@@ -41,32 +34,34 @@ const validateAndNormalizeUrls = (urls: UrlString[]): UrlString[] => {
  * 2. Converting to string then normalizing ensures consistent format
  * 3. validateAndNormalizeUrls removes any potential duplicates and invalid URLs
  */
-const DEFAULT_CONNECTION_URLS = (() => {
+const serverConnectionLibrary = (() => {
   const rawUrls = defaultEnsNodeUrls().map((url) => url.toString());
   const validatedUrls = validateAndNormalizeUrls(rawUrls);
 
   // Guarantee at least 1 URL - this should never happen due to defaultEnsNodeUrls validation
   // but adding as a safety net to maintain the invariant
   if (validatedUrls.length === 0) {
-    throw new Error("DEFAULT_CONNECTION_URLS must contain at least one valid URL");
+    throw new Error("ServerConnectionLibrary must contain at least one valid URL");
   }
 
   return validatedUrls;
 })();
 
-const CONNECTION_PARAM_KEY = "connection";
-
 function _useAvailableENSNodeConnections() {
   const hydrated = useHydrated();
+  const router = useRouter();
   const searchParams = useSearchParams();
   const currentConnection = searchParams.get(CONNECTION_PARAM_KEY);
```

Comment:
```suggestion
  const rawSelectedConnection = searchParams.get(CONNECTION_PARAM_KEY);
```

I assume that's fair?